utils/zerovec/derive/src/varule.rs

```diff
@@ -416,6 +416,26 @@ fn make_encode_impl(
                 dst[#remaining_offset..].copy_from_slice(last_bytes);
             }
         }
+
+        // This second impl exists to allow for using EncodeAsVarULE without cloning
+        //
+        // A blanket impl cannot exist without coherence issues
+        unsafe impl #maybe_lt_bound zerovec::ule::EncodeAsVarULE<#ule_name> for &'_ #name #maybe_lt_bound {
```

Comment:
I realized that another thing preventing us from having separate routes for VarULE and encoded types is that we can only have a single From impl then.

I am warming up to the idea of having separate impls, but I'm not sure how we'd tackle that part.